### PR TITLE
unpinned pymc version

### LINKED ISSUE #936: Run tests with dependency changes
No description provided.

### PR DESCRIPTION
**Pull request by wd60622**
## Description

Checking to see if the tests pass

## Related Issue

- [ ] Closes #936
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--840.org.readthedocs.build/en/840/

### CODE REVIEW

**Comment by wd60622** [comment]
Ref to what fails when no pymc cap @juanitorduz @ColtAllen 

`make_node` method seems to be main issue. The tests with oldest supported pymc werent run due to auto stopping but should pass to my understanding. 

As we eventually work through this change, would it make sense to support current oldest and latest? Or do we expect to see a jump in the pymc min version? Wdyt?

**Comment by juanitorduz** [comment]
Thanks @wd60622 ! I think we could tackle this issue once 5.17 is released and support from 5.16 onwards (so that we support two versions) WDYT?

**Comment by wd60622** [comment]
> Thanks @wd60622 ! I think we could tackle this issue once 5.17 is released and support from 5.16 onwards (so that we support two versions) WDYT? 

Probably the easiest. Alternative would be to do conditional classes based on the version of pymc?

**Comment by juanitorduz** [comment]
> > Thanks @wd60622 ! I think we could tackle this issue once 5.17 is released and support from 5.16 onwards (so that we support two versions) WDYT?
> 
> Probably the easiest. Alternative would be to do conditional classes based on the version of pymc?

No, I think this will be hard to maintain IMO

File: pyproject.toml [hunk 1]
```diff
@@ -32,7 +32,7 @@ dependencies = [
     "pandas",
     "pydantic>=2.1.0",
     # NOTE: Used as minimum pymc version with ci.yml `OLDEST_PYMC_VERSION`
-    "pymc>=5.13.0,<5.16.0",
+    "pymc>=5.13.0",
```
Comment (juanitorduz) [inline pyproject.toml:35]: I think we should always maintain an upper bound. WDYT?

**Comment by wd60622** [comment]
> > > Thanks @wd60622 ! I think we could tackle this issue once 5.17 is released and support from 5.16 onwards (so that we support two versions) WDYT?
> > 
> > 
> > Probably the easiest. Alternative would be to do conditional classes based on the version of pymc?
> 
> No, I think this will be hard to maintain IMO

Let's wait until 5.17 is released then and do a new lower bound of >=5.16.0

**Pull request closed without merging**